spec/models/uploaded_file_spec.rb

```diff
@@ -0,0 +1,121 @@
+    it 'cannot handle S3 URI' do
+      # if the following trips, it means we *might* be able to use just carrierwave again, w/o carrierwave-aws
+      expect(sfile.filename).not_to eq 'image.png'
+    end
+  end
+
+  describe CarrierWave::Storage::File do
+    it 'has default config' do
+      expect(config.storage).to eq described_class # in dev/test
+    end
+    it_behaves_like 'Regular upload'
+    it 'returns a SanitizedFile' do
+      expect(upload.file.file).to be_a CarrierWave::SanitizedFile
+    end
+  end
+
+  # AWSFile constuctor args look like:
+  # uploader:
+  #   <Hyrax::UploadedFileUploader:0x007fcae3312288
+  #     @model=#<Hyrax::UploadedFile id: 3, file: "world.png", user_id: nil, file_set_uri: nil, created_at: "2017-03-02 21:51:45", updated_at: "2017-03-02 21:51:45">,
+  #     @mounted_as=:file, @cache_id="1488491505-93525-0003-2232", @filename="world.png", @original_filename="world.png",
+  #     @file=#<CarrierWave::SanitizedFile:0x007fcae4839ee0 @file="/Users/atz/repos/hyku/public/uploads/tmp/1488491505-93525-0003-2232/world.png", @original_filename="world.png", @content_type=nil>,
+  #     @cache_storage=#<CarrierWave::Storage::File:0x007fcae48383d8 @uploader=#<Hyrax::UploadedFileUploader:0x007fcae3312288 ...>
+  #   >,
+  #   @versions={},
+  #   @storage=#<CarrierWave::Storage::AWS:0x007fcadacbd408 @uploader=#<Hyrax::UploadedFileUploader:0x007fcae3312288 ...>, @connection=#<Aws::S3::Resource>>
+  # >
+  # connection: <Aws::S3::Resource>
+  # path: "/Users/atz/repos/hyku/tmp/uploads/hyrax/uploaded_file/file/3/world.png"
+
+  # With aws wired in, without stubbing, we would get failures telling us to:
+  #   stub_request(:get, "http://169.254.169.254/latest/meta-data/iam/security-credentials/")
+  #     .with(:headers => {'Host'=>'169.254.169.254:80', 'User-Agent'=>'excon/0.55.0'})
+  #     .to_return(:status => 200, :body => 'AWS_DEFAULT_REGION', :headers => {})
+  #   stub_request(:get, "http://169.254.169.254/latest/meta-data/placement/availability-zone/") ...
+  #   stub_request(:get, "http://169.254.169.254/latest/meta-data/iam/security-credentials/AWS_DEFAULT_REGION") ...
+
+  describe CarrierWave::Storage::AWS do
+    let(:file) do
+      # In Controller each file is like:
+      ActionDispatch::Http::UploadedFile.new(
+        tempfile: Tempfile.new,
+        filename: 'world.png',
+        content_type: 'image/png',
+        headers: "Content-Disposition: form-data; name=\"files[]\"; filename=\"world.png\"\r\nContent-Type: image/png\r\nContent-Length: 4218\r\n"
+      )
+    end
+
+    let!(:config) do
+      # reproduce initializer, since it is too late to trigger it by mocking Settings
+      CarrierWave.configure do |config|
+        # config.fog_provider = 'fog/aws'
+        # config.fog_credentials = {
+        #   provider: 'AWS',
+        #   use_iam_profile: true
+        # }
+        config.storage = :aws
+        config.aws_bucket = 'hyku-carrierwave-test'
```

Comment:
Ideally the name of the test bucket would migrate up to be set as a config. If that won't work it should at least be set in a local variable here so the literal bucket name value doesn't need to be repeated in several places. This will make it easier for anyone who forks but wants to swap this out to use their own bucket.